example/generic_social_network/app/resources/follow_resource.py

```diff
@@ -0,0 +1,45 @@
+from cartographer.field_types import StringAttribute, SchemaRelationship
+from cartographer.parsers.schema_parser import SchemaParser
+from cartographer.resources.api_resource import APIResource
+from cartographer.schemas.schema import Schema
+from cartographer.serializers import SchemaSerializer
+from generic_social_network.app.models.tables.follow import Follow
+
+
+class FollowSchema(Schema):
+    SCHEMA = {
+        'type': 'follow',
+        'id': StringAttribute(resource_method='follow_id'),
+        'relationships': {
+            'follower': SchemaRelationship(model_type='user', id_attribute='follower_id'),
+            'followed': SchemaRelationship(model_type='user', id_attribute='followed_id')
+        }
+    }
+
+
+class FollowSerializer(SchemaSerializer):
+    @classmethod
+    def schema(cls):
+        return FollowSchema
+
+    def follow_id(self):
+        return "{follower_id}-{followed_id}".format(
+            follower_id=self.model.follower_id,
+            followed_id=self.model.followed_id
+        )
+
+
+class FollowParser(SchemaParser):
+    @classmethod
+    def schema(cls):
+        return FollowSchema
+
+
+class FollowResource(APIResource):
+    SCHEMA = FollowSchema
+    SERIALIZER = FollowSerializer
+    PARSER = FollowParser
+    # MASK = FollowMask
+    MODEL = Follow
+
+FollowResource.register()
```

Comment:
This is very cool. Agreed that it's much more sane. 

Could we go a step further and create factory methods for cases like `FollowParser` here, where the class is defined only for the sake of returning `FollowSchema`? It seems like this is the common case throughout this project and I think it would also be useful for `patreon_py`.